server/src-lib/Hasura/RQL/IR/Returning.hs

```diff
@@ -79,3 +79,18 @@ hasNestedFld = \case
       AFObjectRelation _ -> True
       AFArrayRelation _  -> True
       _                  -> False
+
+data MutationCTE
```

Comment:
These need to be documented. Maybe a note describing how we validate the check constraint and reference it?